Doist/typist
# feat: Allow all marks to coexist with the Code mark

## Overview

This allows other marks to co-exist with the inline `code` mark, allowing it to be rendered with formatting and links, like this: [**`click me`**](https://todoist.com)

## PR Checklist

Closes https://github.com/Doist/typist/issues/305


## Test plan

- Open the deploy preview link below, and go to the rich text kit's demo
- Paste this content into the editor using cmd + shift + v (paste markdown):
```
[Testing `some code` in a link](https://google.ca/)
```
- [ ] The entire line is wrapped in the link, and the inline block is rendered properly
- [ ] Type some text as an inline code, then use the formatting toolbar to apply bold, italics, and a link to it. Each option is applied correctly
- [ ] Select your linked inline code and its surrounding content, then hit cmd + shift + c. Paste what you've copied somewhere and ensure the resulting markdown is correct

## Demo

| Before                      | After                      |
| --------------------------- | -------------------------- |
| ![image](https://github.com/Doist/typist/assets/8531248/b34cb3f9-5a2c-4546-a905-21be2ea1db0d) | ![image](https://github.com/Doist/typist/assets/8531248/d1815c95-3af0-47c7-882d-13a27b46e17b) |
